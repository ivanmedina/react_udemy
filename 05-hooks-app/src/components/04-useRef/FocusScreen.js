import React, {useRef} from 'react'

export const FocusScreen = () => {

    const inputRef = useRef();

    const handleClick=()=>{
        inputRef.current.select();
        console.log(inputRef);
    }


    return (
        <div>
            <h1>FocusScreen</h1>
            <hr/>
            <input
                ref={inputRef}
                className="form-cotrol"
                placeholder="su nombre"
            />
            <button
                className="btn btn-outline-primary mt-5"
                onClick={handleClick}
            >
                Focus
            </button>
        </div>
    )
}
