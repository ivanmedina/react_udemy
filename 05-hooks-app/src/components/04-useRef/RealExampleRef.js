import React ,{useState}from 'react'
import { MultipleCustomHooks } from "../03-examples/MultipleCustomHooks";

export const RealExampleRef = () => {

    const [show, setShow] = useState(false);
    return (
        <div>
            <h1>Real Example Ref</h1>
            <hr/>

            {show && <MultipleCustomHooks />}

            <button
                className="btn btn-primary"
                onClick={
                    ()=>{
                        setShow( !show);
                    }
                }
            >
                Show/hide
            </button>
        </div>
    )
}
