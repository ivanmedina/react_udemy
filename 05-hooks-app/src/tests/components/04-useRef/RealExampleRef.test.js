import React from 'react';
import {shallow} from 'enzyme';
import {RealExampleRef} from '../../../components/04-useRef/RealExampleRef'


describe('Prueba en <RealExampleRef />',()=>{

    const wrapper=shallow(<RealExampleRef />)
    test('debe de mostrarse correctamente',()=>{
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de mostrarse el componente <MUltipleCustomHooks/>',()=>{
        wrapper.find('button').simulate('click');
        expect(wrapper.find('MultipleCustomHooks').exists() ).toBe(true);
    })

})