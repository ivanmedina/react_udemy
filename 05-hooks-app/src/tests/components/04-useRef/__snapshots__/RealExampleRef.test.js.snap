// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <RealExampleRef /> debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    Real Example Ref
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Show/hide
  </button>
</div>
`;
